src/transformers/tokenization_utils.py

```diff
@@ -1958,75 +2295,86 @@ def _convert_encoding(
             if return_offsets_mapping:
                 encoding_dict["offset_mapping"].append(e.offsets)
 
-        # Prepare inputs as tensors if asked
-        if return_tensors == "tf" and is_tf_available():
-            encoding_dict["input_ids"] = tf.constant(encoding_dict["input_ids"])
-            if "token_type_ids" in encoding_dict:
-                encoding_dict["token_type_ids"] = tf.constant(encoding_dict["token_type_ids"])
-
-            if "attention_mask" in encoding_dict:
-                encoding_dict["attention_mask"] = tf.constant(encoding_dict["attention_mask"])
-
-        elif return_tensors == "pt" and is_torch_available():
-            encoding_dict["input_ids"] = torch.tensor(encoding_dict["input_ids"])
-            if "token_type_ids" in encoding_dict:
-                encoding_dict["token_type_ids"] = torch.tensor(encoding_dict["token_type_ids"])
-
-            if "attention_mask" in encoding_dict:
-                encoding_dict["attention_mask"] = torch.tensor(encoding_dict["attention_mask"])
-        elif return_tensors is not None:
-            logger.warning(
-                "Unable to convert output to tensors format {}, PyTorch or TensorFlow is not available.".format(
-                    return_tensors
-                )
-            )
+        if return_tensors is not None:
+            for key, value in encoding_dict.items():
+                if return_tensors == "tf" and is_tf_available():
+                    encoding_dict[key] = tf.constant(value)
+                elif return_tensors == "pt" and is_torch_available():
+                    encoding_dict[key] = torch.tensor(value)
+                elif return_tensors is not None:
+                    logger.warning(
+                        "Unable to convert output to tensors format {}, "
+                        "PyTorch or TensorFlow is not available.".format(return_tensors)
+                    )
 
         return encoding_dict
 
-    def _convert_token_to_id_with_added_voc(self, token):
-        id = self._tokenizer.token_to_id(token)
-        if id is None:
+    def _convert_token_to_id_with_added_voc(self, token: int) -> str:
+        index = self._tokenizer.token_to_id(token)
+        if index is None:
             return self.unk_token_id
-        return id
+        return index
 
-    def _convert_id_to_token(self, index: int) -> str:
+    def _convert_id_to_token(self, index: int) -> Optional[str]:
         return self._tokenizer.id_to_token(int(index))
```

Comment:
Is expl. casting to `int` necessary here 🤔

Reply:
Safer to do it this way, in the past, we've seen issues when we weren't too explicit, for example, SentencePiece tokenizers crashing because the inputs were NumPy integers rather than standard python integers